### PR TITLE
[WIP] [Epic] Susskartoffel -- Revocations and policy info serializations

### PR DESCRIPTION
**Pull request by tuxxy**
WIP

### CODE REVIEW

File: nucypher/policy/collections.py [hunk 1]
```diff
@@ -209,6 +211,67 @@ def __len__(self):
...
+    @classmethod
+    def from_json(cls, data: str):
+        """
+        Deserializes the PolicyCredential from JSON.
+        """
+        cred_json = json.loads(data)
+
+        alice_verifying_key = UmbralPublicKey.from_bytes(
+                                    cred_json['alice_verifying_key'],
+                                    decoder=bytes().fromhex)
```
Comment (KPrasch) [inline nucypher/policy/collections.py]: Extra set of parenthesis after `bytes` here - passing the `fromhex` method bound to an empty bytestring.

File: nucypher/policy/policies.py [hunk 2]
```diff
@@ -277,6 +277,22 @@ def publish(self, network_middleware: RestMiddleware) -> dict:
         """
         return self.publish_treasure_map(network_middleware=network_middleware)
 
+    def credential(self, with_treasure_map=True):
```
Comment (KPrasch) [inline nucypher/policy/policies.py]: I think `make_credential` or similar is more descriptive here, since we are not *using* the credential here.

**Comment by michwill** [comment]
I think, rather than returning a credential _with_ every grant, we need to make the output of `grant` being a credential (there is not much difference: just slightly more info in the credential)

File: nucypher/policy/policies.py (hunk 2 above)
Comment (michwill) [reply nucypher/policy/policies.py]: I like more concise option. Or what, in your opinion, does the `credential()` method name say intuitively?

**Comment by michwill** [comment]
Hodl on, I think repearing the same info in credential _and_ grant output is ugly. Refactoring a bit

File: nucypher/policy/collections.py (hunk 1 above)
Comment (tuxxy) [reply nucypher/policy/collections.py]: Not an extra set, this is intentional. :)

File: nucypher/policy/policies.py (hunk 2 above)
Comment (tuxxy) [reply nucypher/policy/policies.py]: Credential is already a verb. When you _credential_ someone, you are giving them credential.

**Comment by tuxxy** [comment]
@michwill we don't return a credential with every grant. Nothing has changed on that, only in the character control is that what happens because we can't return a `Policy` object.

In the Python API, we still only return a `policy`, which you can call `policy.credential()` to have the credential.

File: nucypher/policy/collections.py (hunk 1 above)
Comment (KPrasch) [reply nucypher/policy/collections.py]: @tuxxy - Why do this / What effect does it have? Why not use the class method directly `bytes.from_hex`?

File: nucypher/policy/policies.py (hunk 2 above)
Comment (KPrasch) [reply nucypher/policy/policies.py]: Using the verb form of `credential` sounds to my ear like you are *presenting it to gain access*, not merely possessing it.  `Policy` can be used to produce a credential, but It sounds  to me like `Alice` / `Bob` do the actual credentialing.

**Comment by michwill** [comment]
Well, ok, then I'd be ok with this one as a temporary measure. If I touch this, I'm afraid I'll be rebuilding a good part of the codebase, heh.

File: nucypher/policy/collections.py (hunk 1 above)
Comment (michwill) [reply nucypher/policy/collections.py]: Yeah, I think `bytes.fromhex` works just as well

**Comment by KPrasch** [comment]
Rebased over `64465d6f7a8536f6d7b52e9d8daa9482c311925b`

**Comment by KPrasch** [comment]
Moved to #1367

**Pull request closed without merging**